### DISCUSSION TITLE
Have a window critera for 'under mouse' or similar

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#5440] [Ideas]
I'm aware that this is likely not super useful for most, but it would make some tasks much easier for me.

Specifically I am trying to resolve an issue I have. Currently I have 'focus_follows_mouse off' to fix an issue with another program that behaves poorly with frequent focus changes. However that introduced a new issue which is now when I try to kill a container sometimes what I think is my focus isn't actually my focus and I kill the wrong thing.

Ideally I would want to have some kind of condition so the kill command only runs on the window under the mouse, not the focused window (or barring that, I could make a command to switch focus to the window under mouse then close it). 

However I see no real way of doing this currently. Best fix I found was to run xprop, sleep for 0.01 seconds, emulate a left click, then use that to find the window to close.

That obviously isn't the ideal way to solve it and has the risk of causing all sorts of odd race conditions. So I'm hoping there is either a better alternative or a new criteria could be added to let me fix the issue properly.

**Comment by orestisfl** [comment]
You can do something like

```
eval $(xdotool getmouselocation --shell)
xdotool windowclose $WINDOW
```
